link/tracing.go

```diff
@@ -0,0 +1,161 @@
+package link
+
+import (
+	"fmt"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal/btf"
+	"github.com/cilium/ebpf/internal/sys"
+)
+
+type tracing struct {
+	RawLink
```

Comment:
I think it's better to have embedded by pointer, (a bit less memory allocation)
```suggestion
	*RawLink
```

Reply:
All of the other link types currently embed non-pointer RawLink, so it's better to be consistent here. I think I had a reason for non-pointer at some point, but I can't remember it now. Probably doesn't make a big difference either way.